### ISSUE TITLE
[Bug]: [7] Couldn't connect to servec

### ISSUE THREAD

**Issue by VehementHam** [flatpak/flatpak#5740]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

Flatpak 1.14.5

### What Linux distribution are you using?

Alpine Linux

### Linux distribution version

Edge

### What architecture are you using?

x86_64

### How to reproduce

I installed Alpine Linux

I followed the instructions on [https://flatpak.org/setup/alpine/](https://flatpak.org/setup/alpine/).

I attempt to use pretty much any of `flatpak`'s functionalities.

### Expected Behavior

Packages are installed/searched, and the appstream is updated.

### Actual Behavior

# Installing a package
```
localhost:~$ doas flatpak install com.play0ad.zeroad
Looking for matches…
Remotes found with refs similar to ‘com.play0ad.zeroad’:

   1) ‘flathub’ (system)
   2) ‘fluthub’ (system)

Which do you want to use (0 to abort)? [0-2]: 2
Required runtime for com.play0ad.zeroad/x86_64/stable (runtime/org.freedesktop.Platform/x86_64/22.08) found in remotes:

   1) fluthub
   2) flathub

Which do you want to install (0 to abort)? [0-2]: 2

com.play0ad.zeroad permissions:
    ipc   network   pulseaudio   x11  dri  shm



        ID                                             Branch                 Op            Remote             Download
 1. [✗] com.play0ad.zeroad.Locale                      stable                 i             fluthub              1.0 kB / 1.0 kB
 2. [✗] org.freedesktop.Platform.GL.default            22.08                  i             flathub              1.0 kB / 148.3 MB
 3. [✗] org.freedesktop.Platform.GL.default            22.08-extra            i             flathub              1.0 kB / 148.3 MB
 4. [✗] org.freedesktop.Platform.Locale                22.08                  i             flathub              1.0 kB / 333.7 MB
 5. [✗] org.freedesktop.Platform.openh264              2.2.0                  i             flathub              1.0 kB / 944.3 kB
 6. [✗] org.freedesktop.Platform                       22.08                  i             flathub              1.0 kB / 214.5 MB
 7. [ ] com.play0ad.zeroad                             stable                 i             fluthub            < 1.8 GB

Warning: While pulling runtime/com.play0ad.zeroad.Locale/x86_64/stable from remote fluthub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Warning: While pulling runtime/org.freedesktop.Platform.GL.default/x86_64/22.08 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Warning: While pulling runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Warning: While pulling runtime/org.freedesktop.Platform.Locale/x86_64/22.08 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Warning: While pulling runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Error: While pulling runtime/org.freedesktop.Platform/x86_64/22.08 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
error: Failed to install org.freedesktop.Platform: While pulling runtime/org.freedesktop.Platform/x86_64/22.08 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
```
# Searching
```
localhost:~$ flatpak search 0ad
No matches found
localhost:~$ 
```
# Updating the appstream
```
localhost:~$ flatpak update --appstream
Updating appstream data for remote flathub
Error updating: While pulling appstream2/x86_64 from remote flathub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
Updating appstream data for remote fluthub
Error updating: While pulling appstream2/x86_64 from remote fluthub: While fetching https://dl.flathub.org/repo/config: [7] Couldn't connect to server
```

### Additional Information

I am using Alpine Linux. Network is functional.

**Comment by chrisawi** [comment]
Are you using a proxy?

Are you able to `curl https://dl.flathub.org/repo/config` ?

**Comment by hfiguiere** [comment]
Maybe your ISP is blocking it. We have had report of this happening.

**Comment by VehementHam** [comment]
> Are you using a proxy?

No.

> Are you able to curl `https://dl.flathub.org/repo/config` ?

Yes.

**Comment by VehementHam** [comment]
> Maybe your ISP is blocking it. We have had report of this happening.

No. Flathub has worked in tha past. I cannot see a possible insentive for blocking Flathub either.

**Comment by hfiguiere** [comment]
Thing is flatpak != flathub. (this is flatpak issue tracker) If any indication there is, it's a problem between your location and flathub.

It's not because it worked in the past that it works now.

**Comment by VehementHam** [comment]
`flatpak update` works on another machine.

**Comment by VehementHam** [comment]
The errors are inconsistant. Sometimes it starts to install, and other times, it immediately reports an error.

**Comment by VehementHam** [comment]
Any idea what I should do?

**Comment by AmrElsayyad** [comment]
Are there any updates on this issue?

**Comment by VehementHam** [comment]
Yes. I am still having the issue.

**Comment by dus2002** [comment]
I had this issue too. I suspected something was wrong with dns and I reinstalled. Just into the bare minimum I tested, without DE, and got the same (exact same) output from both regular and beta repos, so I removed my first resolv.conf entry (my router) and issued "nameserver 8.8.8.8". Now the problem is gone. Everything in flatpak connects. Alpine Edge updated to the time of posting
![IMG_20240816_195124_999](https://github.com/user-attachments/assets/ba039f96-44f6-4ce1-b633-07c42c3c4785)

**Comment by smcv** [comment]
> I get the same error on my other Alpine machine. Not on machines running other operating systems.

> I had this issue too. … Alpine Edge

Because this seems to be specific to Alpine Linux Edge, it looks like a problem with the way Flatpak has been packaged in Alpine, or a problem with dependencies provided by Alpine (for example it could be a problem with libcurl, TLS CA certificates, name resolution, or the C library more generally).

Am I correct to think that Alpine uses musl as its C library, unlike most Linux distributions which use glibc? If that's the case, then that's an obvious source of possible incompatibilities.

**Comment by smcv** [comment]
I would suggest talking to Alpine's support channels, specifically the packagers responsible for its `flatpak` package.

**Comment by VehementHam** [comment]
https://gitlab.alpinelinux.org/alpine/aports/-/issues/15915

**Comment by jusko** [comment]
> I had this issue too. I suspected something was wrong with dns and I reinstalled. Just into the bare minimum I tested, without DE, and got the same (exact same) output from both regular and beta repos, so I removed my first resolv.conf entry (my router) and issued "nameserver 8.8.8.8". Now the problem is gone. 

Same. Spent half a day hitting the wall on this because all other networking (ping, curl, wget) works. Your comment on DNS made me try nuking my router out of /etc/resolve.conf too (I have experienced it to be a bit flakey at times). Problem solved. 

Flatpak appears to be resolving in an obscure way and there's no way to know from its output. 

Thank you for commenting @dus2002 ---solved a very painful problem!
